src/OmniSharp.LanguageServerProtocol/Handlers/HoverHandler.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using OmniSharp.Extensions.LanguageServer.Capabilities.Client;
+using OmniSharp.Extensions.LanguageServer.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Mef;
+using OmniSharp.Models.TypeLookup;
+
+namespace OmniSharp.LanguageServerProtocol.Handlers
+{
+    [Shared, Export(typeof(HoverHandler))]
+    class HoverHandler : IHoverHandler
+    {
+        private HoverCapability _capability;
+        private readonly IRequestHandler<TypeLookupRequest, TypeLookupResponse> _definitionHandler;
+        private readonly DocumentSelector _documentSelector;
+
+        [ImportingConstructor]
+        public HoverHandler(IEnumerable<IRequestHandler> handlers, DocumentSelector documentSelector)
+        {
+            _definitionHandler = handlers.OfType<IRequestHandler<TypeLookupRequest, TypeLookupResponse>>().Single();
```

Comment:
How is multiple languages supported? E.g. in the Cake PR we register a second handler of this type.

Reply:
I came to that realization last night, and added a comment about changing that around.  I'll be thinking about that today and probably figure out a decent solution fairly soon.

The note wasn't near that code however https://github.com/OmniSharp/omnisharp-roslyn/blob/f9af5e2ec15b56aced4afd590c5fb00dacb6d1a0/src/OmniSharp.LanguageServerProtocol/LanguageServerHost.cs#L105-L107

Reply:
Great! This is how we currently resolve `C#` version of the handler we are overriding in Cake https://github.com/cake-build/omnisharp-roslyn/blob/da3b227582f949e2c378d6280ab0b237811228a2/src/OmniSharp.Cake/Services/RequestHandlers/CakeRequestHandler.cs#L30-L45